src/java.base/share/classes/javax/crypto/spec/snippet-files/PackageSnippets.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import javax.crypto.Cipher;
+import javax.crypto.spec.HPKEParameterSpec;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.util.Arrays;
+
+class PackageSnippets {
+    public static void main(String[] args) throws Exception {
+        // @start region="hpke-spec-example"
+        // Key pair generation
+        KeyPairGenerator g = KeyPairGenerator.getInstance("X25519");
+        KeyPair kp = g.generateKeyPair();
+
+        // The HPKE sender side is initialized with the recipient's public key
+        // and default HPKEParameterSpec with an application-supplied info.
+        Cipher sender = Cipher.getInstance("HPKE");
+        HPKEParameterSpec ps = HPKEParameterSpec.of()
+                .info("app_info".getBytes(StandardCharsets.UTF_8));
+        sender.init(Cipher.ENCRYPT_MODE, kp.getPublic(), ps);
+
+        // Retrieve the actual parameters used from the sender.
```

Comment:
I think it would be more clear if you didn't name the cipher objects `sender` and `recipient` because there can be confusion as to whether you mean the cipher objects or the sender/receiver entities.

Reply:
I'll named them `senderCipher` and `recipientCipher`.